py/torch_tensorrt/dynamo/conversion/impl/pad.py

```diff
@@ -19,43 +27,36 @@ def constant_padNd(
     pad: Sequence[int],
     value: Union[int, float] = 0,
 ) -> TRTTensor:
-    """
-    Note: IPaddingLayer is deprecated in TensorRT 8.2 and will be removed in TensorRT 10.0.
-    Use ISliceLayer to pad the tensor, which supports new non-constant, reflects padding
-    mode and clamp, and supports padding output with dynamic shape.
-    """
     if has_dynamic_shape(input.shape):
         assert input.shape[1] != -1, "Channel dim can't be dynamic for padding."
 
-    # Implement constant padding via concat
-    curr_dim = len(input.shape) - 1
-
-    for i in range(0, len(pad), 2):
-        input_shape = list(input.shape)
-
-        pre_pad = pad[i]
-        post_pad = pad[i + 1]
-        pre_pad_shape = copy.deepcopy(input_shape)
-        pre_pad_shape[curr_dim] = pre_pad
-        pre_pad_tensor = torch.full(pre_pad_shape, float(value))
-        if pre_pad == post_pad:
-            post_pad_tensor = pre_pad_tensor
-        else:
-            post_pad_shape = copy.deepcopy(input_shape)
-            post_pad_shape[curr_dim] = post_pad
-            post_pad_tensor = torch.full(post_pad_shape, float(value))
-        output = impl.cat.cat(
-            ctx,
-            target,
-            source_ir,
-            f"{name}_concat{curr_dim}",
-            input=(pre_pad_tensor, input, post_pad_tensor),
-            dim=curr_dim,
+    rank = len(input.shape)
+
+    if len(pad) / 2 > rank:
```

Comment:
Should this be a `//` instead of `/`?

Reply:
Both are fine because `len(pad)` should be even. BTW, if PyTorch checks some restrictions like `len(pad) should be even`, do we need to check again within our converters?

Reply:
It shouldn't be necessary if PyTorch checks it, but can't hurt to add the check

Reply:
Additionally, `//` is preferred if the results shouldn't matter, since `//` will output an integer